File: app/src/main/java/com/sumalogos/webster/sumalogos/util/APIClient.java
package com.sumalogos.webster.sumalogos.util;

import java.util.concurrent.TimeUnit;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;
import retrofit2.Retrofit;
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory;
import retrofit2.converter.gson.GsonConverterFactory;

/**
 * Created by webster on 18/01/18.
 */

public class APIClient {
    public static Retrofit getClient() {
        Retrofit retrofit;

        HttpLoggingInterceptor interceptor = new HttpLoggingInterceptor();
        interceptor.setLevel(HttpLoggingInterceptor.Level.BODY);
        OkHttpClient client = new OkHttpClient.Builder()
                .connectTimeout(AppConstant.TIME_OUT_LIMIT, TimeUnit.SECONDS)
                .writeTimeout(AppConstant.TIME_OUT_LIMIT, TimeUnit.SECONDS)
                .readTimeout(AppConstant.TIME_OUT_LIMIT, TimeUnit.SECONDS)
                .hostnameVerifier((hostname, session) -> true)
                .build();

        retrofit = new Retrofit.Builder()
                .baseUrl(AppConstant.BASE_API_URL)
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                .client(client)
                .build();

        return retrofit;
    }
}

File: app/src/main/java/com/sumalogos/webster/sumalogos/util/APIClient.kt
package com.sumalogos.webster.sumalogos.util

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor
import retrofit2.Retrofit
import retrofit2.adapter.rxjava2.RxJava2CallAdapterFactory
import retrofit2.converter.gson.GsonConverterFactory
import java.util.concurrent.TimeUnit


/**
 * Created by webster on 01/01/18.
 */

object APIClient {
    val client: Retrofit
        get() {
            val retrofit: Retrofit

            val interceptor = HttpLoggingInterceptor()
            interceptor.level = HttpLoggingInterceptor.Level.BODY
            val client = OkHttpClient.Builder()
                    .connectTimeout(AppConstant.TIME_OUT_LIMIT.toLong(), TimeUnit.SECONDS)
                    .writeTimeout(AppConstant.TIME_OUT_LIMIT.toLong(), TimeUnit.SECONDS)
                    .readTimeout(AppConstant.TIME_OUT_LIMIT.toLong(), TimeUnit.SECONDS)
                    .hostnameVerifier { _, _ -> true }
                    .build()

            retrofit = Retrofit.Builder()
                    .baseUrl(AppConstant.BASE_API_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .addCallAdapterFactory(RxJava2CallAdapterFactory.create())
                    .client(client)
                    .build()

            return retrofit
        }
}